src/states/common/base.rs

```diff
@@ -354,28 +361,30 @@ pub trait Base: Display {
         None
     }
 
-    fn send_direct_message(&mut self, dst: &ConnectionInfo, content: Variant) {
-        let message = if let Ok(message) = self.to_signed_direct_message(content) {
-            message
-        } else {
-            return;
+    fn send_direct_message(&mut self, recipient: &ConnectionInfo, variant: Variant) {
+        let message = match Message::single_src(self.full_id(), DstLocation::Direct, variant) {
+            Ok(message) => message,
+            Err(error) => {
+                error!("{} - Failed to create message: {:?}", self, error);
+                return;
+            }
         };
 
-        let message = match to_network_bytes(&message) {
-            Ok(bytes) => bytes,
+        let bytes = match serialize(&message) {
```

Comment:
Not 100% sure that removing `to_network_bytes`... makes things clearer. It was quite good to see that this was serialization of a specific object we send on the wire. If we want to do this change, maybe should be in its own PR, but I'm not yet convinced we want to.